chainstate/src/detail/tests/mod.rs

```diff
@@ -100,38 +92,78 @@ fn chainstate_with_config(
     .unwrap()
 }
 
-fn produce_test_block(prev_block: &Block, orphan: bool) -> Block {
-    produce_test_block_with_consensus_data(prev_block, orphan, ConsensusData::None)
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TestBlockInfo {
```

Comment:
What problem does this solve?

Reply:
The `generate_block` function takes the parent block as a parameter. However, the parent can be genesis as well so it can't just take the block directly. This struct collects just bits and pieces that are needed to generate the next block (i.e. parent's ID and outputs), and can be constructed from normal blocks or from genesis (or manually if desired).

It would be much nicer to have proper UTXO set abstraction in the code, which would abstract the user away from the consideration where the output comes from (tx output, block reward, genesis premine). I believe it would make some parts of the node code nicer too. This is just hacked-up poor man's version of it for testing.